pkg/connector/login.go

```diff
@@ -0,0 +1,188 @@
+					Required: true,
+					Sources: []bridgev2.LoginCookieFieldSource{
+						{Type: bridgev2.LoginCookieTypeCookie, Name: "ct0"},
+					},
+				},
+				{
+					ID:       "auth_token",
+					Required: true,
+					Sources: []bridgev2.LoginCookieFieldSource{
+						{Type: bridgev2.LoginCookieTypeCookie, Name: "auth_token"},
+					},
+				},
+				/*
+						{
+							ID: "guest_id",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "guest_id" },
+							},
+						},
+						{
+							ID: "twid",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "twid" },
+							},
+						},
+						{
+							ID: "kdt",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "kdt" },
+							},
+						},
+						{
+							ID: "night_mode",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "night_mode" },
+							},
+						},
+						{
+							ID: "personalization_id",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "personalization_id" },
+							},
+						},
+						{
+							ID: "guest_id_marketing",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "guest_id_marketing" },
+							},
+						},
+						{
+							ID: "guest_id_ads",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "guest_id_ads" },
+							},
+						},
+						{
+							ID: "d_prefs",
+							Required: false,
+							Sources: []bridgev2.LoginCookieFieldSource{
+								{ Type: bridgev2.LoginCookieTypeCookie, Name: "d_prefs" },
+							},
+						},
+					},
+				*/
+			},
+		},
+	}, nil
+}
+
+func (t *TwitterLogin) Cancel() {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (t *TwitterLogin) SubmitCookies(ctx context.Context, cookies map[string]string) (*bridgev2.LoginStep, error) {
+	cookieStruct := twitCookies.NewCookies(cookies)
+	meta := &UserLoginMetadata{
+		Cookies: cookieStruct.String(),
+	}
+
+	clientOpts := &twittermeow.ClientOpts{
+		Cookies:       cookieStruct,
+		WithJOTClient: true,
+	}
+	client := twittermeow.NewClient(clientOpts, t.User.Log)
+
+	_, _, err := client.LoadMessagesPage()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load messages page after submitting cookies")
+	}
+
+	id := networkid.UserLoginID(client.GetCurrentUserID())
+	ul, err := t.User.NewLogin(
+		ctx,
+		&database.UserLogin{
+			ID:       id,
+			Metadata: meta,
+		},
+		&bridgev2.NewLoginParams{
+			DeleteOnConflict:  true,
+			DontReuseExisting: false,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &bridgev2.LoginStep{
```

Comment:
The client inside `ul` needs to be connected manually (it may also make sense to reuse the client that was created above, that can be done via `NewLoginParams`)

Reply:
> Can you clarify please? Is it by providing the LoadUserLogin function?

Yeah that's the way to provide an existing client

> As for manual connection, it does not seem like it actually needs the connection. Looks like the LoadMessagesPage endpoint is invoked here to get more auth information (like getting the verification token)

I'd assume it needs a call to start polling at least after it's refactored to not start automatically (because `LoadUserLogin` isn't allowed to start polling)